cpu/numactl.py

```diff
@@ -75,3 +104,106 @@ def test(self):
                 self.log.warn('Few tests failed due to less NUMA mem-nodes')
             else:
                 self.fail('test failed, Please check debug log')
+
+    def get_nodes_with_cpus(self):
+        '''
+        Returns a dictonary, 
+        1. node number as key and 
+        2. associated CPUS as values. 
+        '''
+
+        cpu_path = '/sys/devices/system/node/has_cpu'
+        if not os.path.exists(cpu_path):
+            raise CpuError("No NUMA nodes have cpu")
+            self.cancel("No NUMA nodes have CPU")
+               
+        if len(genio.read_file(cpu_path).rstrip('\n')) < 2:
+            self.cancel("Required atleast two NUMA nodes with CPU assigned  for this test case!")
+
+        self.dict1 = {}
+        for path in glob.glob("/sys/devices/system/node/node[0-9]*"):
+            self.node= (int(re.search(r".node(\d+)$",path).group(1)))
+            self.node_dir = (path + "/")
+            self.pinned_cpus = []
+            for cpu_numbers in glob.glob(self.node_dir+"cpu[0-9]*"):
+                self.cpu = (int(re.search(r".cpu(\d+)$",cpu_numbers).group(1)))
+                self.pinned_cpus.append(self.cpu)
+            self.dict1[self.node] = (sorted(self.pinned_cpus))
+               
+        return (self.dict1)
+
+
+    def numa_ping(self,cmd):
+        '''
+        Ping Test to remote Peer with -f and count options.
+        '''
+        output = process.run(cmd, shell=True,ignore_status=True).stdout.decode("utf-8").split(",")
+        if " 0% packet loss"  not in output:
+            self.cancel("failed due to packet loss")
```

Comment:
Good to move this to utils

Reply:
As discussed, this is not much used out side of this test.. lets keep it here it self